### PR TITLE
multi: replace occurrences of "LSAT" to "L402"

### PR DESCRIPTION
**Pull request by starius**
# Description

Aperture was updated to include https://github.com/lightninglabs/aperture/pull/135 . Occurrences of "LSAT" in the loop's code were replaced with "L402".

This PR depends on https://github.com/lightninglabs/aperture/pull/135 . Commit "update aperture to include lsat to l402 renaming" should be updated after https://github.com/lightninglabs/aperture/pull/135 is merged.

Some unrelated changes:

  * looprpc,swapserverrpc: update Go image to 1.19.10 (needed for the following item)
  * go: bump protobuf to v1.33.0-hex-display (needed to update to the recent aperture version)
  * multi: apply make fmt

#### Pull Request Checklist
- [x] Update `release_notes.md` if your PR contains major features, breaking changes or bugfixes

This is a breaking change!

In loopd.conf file `maxlsatcost` and `maxlsatfee` were renamed to `maxl402cost` and `maxl402fee` accordingly. If they have been changed locally, the file has to be updated for loopd to recognize the options.

The path in looprpc "/v1/lsat/tokens" was renamed to "/v1/l402/tokens" and the corresponding method was renamed from `GetLsatTokens` to `GetL402Tokens`. Update `loop` and `loopd` simultaneously otherwise this RPC won't work.

### CODE REVIEW

**Review by hieblmi — APPROVED** [review]
Thank you very much for the PR, tACK from my side. I tested your changes on my static address branch which fetches an L402 from a local instance. Don't see any issues. The only remaining `lsat` references in code are in release notes, so LGTM

**Comment by hieblmi** [comment]
On another note, we should check if that these changes don't break the server.

File: looprpc/Dockerfile
Comment (guggero) [inline looprpc/Dockerfile:1]: Here's an example for the changes required to the `protobuf-compiler` and `clang-format` versions if you wanted to go to a `-bookworm` (Go 1.2x) version: https://github.com/lightninglabs/taproot-assets/blob/main/taprpc/Dockerfile#L1
It's not too involved, so I would suggest doing it.

File: go.mod
Comment (guggero) [inline go.mod:208]: Reminder to not merge this PR until the dependent PR was merged.

File: loopd/config.go
Comment (guggero) [inline loopd/config.go:163]: This might cause users that have the old flag set to not be able to start their daemon after upgrading. Should we keep the old flags but set them to `hidden:true` and log a deprecation warning if they are set? Then remove the actual flags in a later release?

File: looprpc/client.yaml
Comment (guggero) [inline looprpc/client.yaml:30]: Should we keep the old mapping (can have `additional_bindings` to map multiple URIs to the same method, see https://github.com/lightninglabs/taproot-assets/blob/main/taprpc/taprootassets.yaml#L66 for example) to not break any client applications?

File: release_notes.md
Comment (guggero) [inline release_notes.md:25]: Hmm, okay, so we warn about breaking changes here. So I guess it's up to the team to decide whether that's enough or we want to keep backward compatibility as suggested in the comments above.

**Review by guggero — COMMENTED** [review]
Code looks good to me. See inline discussion about potential backward compatibility fixes.

Going to approve once dependent PR is merged to avoid merging prematurely.

File: looprpc/Dockerfile
Comment (starius) [reply looprpc/Dockerfile]: Fixed

File: go.mod
Comment (starius) [reply go.mod]: Fixed

File: loopd/config.go
Comment (starius) [reply loopd/config.go:165]: The flags were re-added in commit "loopd: recorgnize maxlsatcost and maxlsatfee flags".
I ran loopd locally to verify that if both old and new variants of an option are used simultaneously, it fails to start; and if only the old variant is provided, it starts with a deprecation warning.

File: looprpc/client.yaml
Comment (starius) [reply looprpc/client.yaml:30]: I added `additional_bindings`.

But it does not provide backward compatibility for GRPC. When I run new loopd with old loop, I get the following error, trying to to run `loop listauth`:
```
[loop] rpc error: code = Unimplemented desc = unknown method GetLsatTokens for service looprpc.SwapClient
```

File: release_notes.md
Comment (starius) [reply release_notes.md:26]: I updated the release notes. Now the only broken scenario that I found is `loop listauth` if the `loop` binary is from a previous release, see above.

File: swapserverrpc/Dockerfile
Comment (guggero) [inline swapserverrpc/Dockerfile:1]: Reminder to push a new tag for the `swapserverrpc` module after merging this PR.

File: looprpc/client.pb.go
Comment (guggero) [inline looprpc/client.pb.go:3]: micro-nit (non-blocking): This should probably go into the previous commit.

File: loopd/perms/perms.go
Comment (guggero) [inline loopd/perms/perms.go:72]: Hmm, I didn't really think about this in my last review. But I think this might break some client applications (like Terminal Web, RTL or others that have Loop integrated) that rely on the old RPC name. To avoid headaches, I recommend to not rename this. Maybe we can add a new RPC and declare this one as deprecated, then remove it in a couple of releases.

File: loopd/utils.go
Comment (guggero) [inline loopd/utils.go:55]: :+1:

File: looprpc/client.yaml
Comment (guggero) [reply looprpc/client.yaml:30]: Yeah, this might be a problem for some web UIs that use the RPC method directly, like Terminal Web (see my comment above).

**Review by guggero — COMMENTED** [review]
Code looks great now, thanks for the updates!
A remaining question around backward compatibility of the RPC interface. Normally we don't break backward compatibility for at least two major releases (at least that's the idea in `lnd` but in practice we end up rarely removing/breaking anything at all)...

File: looprpc/client.pb.go
Comment (starius) [reply looprpc/client.pb.go:3]: This is actually updated in this commit.

File: loopd/perms/perms.go
Comment (starius) [reply loopd/perms/perms.go:72]: I added GetLsatTokens back as a wrapper. Updated release notes file.

Tested with old `loop` and new `loopd`:

```
$ loop listauth
[]
```

Loopd log:

```
2024-04-25 13:52:27.367 [WRN] LOOPD: Received deprecated call GetLsatTokens. Please update the client software. Calling GetL402Tokens now.
```

File: looprpc/client.proto
Comment (guggero) [inline looprpc/client.proto:83]: I think we can remove the `loop: listauth` part, as that's for the API docs generator only, which should use the new RPC.

**Review by guggero — APPROVED** [review]
Nice, LGTM :tada:

File: looprpc/client.proto
Comment (starius) [reply looprpc/client.proto]: Fixed

**Pull request merged**
